caikit/runtime/http_server.py

```diff
@@ -311,35 +311,13 @@ def _get_request_params(
         # convert pydantic objects to our DM objects
         for param_name, param_value in request_params.items():
             if issubclass(type(param_value), pydantic.BaseModel):
-                request_params[param_name] = self.build_dm_object(param_value)
+                request_params[param_name] = self._build_dm_object(param_value)
         return request_params
 
-    def build_dm_object(self, pydantic_model: pydantic.BaseModel) -> DataBase:
```

Comment:
No code change, just converted to `classmethod` and renamed it to `_build_dm_object`

Reply:
Nice!

It's looking like it's hight time for some restructuring here- this one file has a whole lotta stuff about `pydantic` and `http handlers` intermixed.

Thoughts on extracting out the pydantic logic to a `caikit/core/data_model/pydantic_view.py` or similar spot? And slapping a few unit tests in there for:
- pydantic model creation from DM class
- pydantic instance -> dm instance translation
- dm instance -> pydantic instance translation